diff --git a/og_subgroups.module b/og_subgroups.module
index f55ba1b..8ff9dd6 100755
--- a/og_subgroups.module
+++ b/og_subgroups.module
@@ -5,12 +5,14 @@
  * Enable defining hierarchy of groups for organic groups.
  */
 
+define('OG_SUBGROUP_REQUIRED_CTOOLS_API', '2.0-alpha');
+
 /**
  * Implementation of hook_ctools_plugin_directory().
  */
 function og_subgroups_ctools_plugin_directory($module, $plugin) {
   // Safety: go away if CTools is not at an appropriate version.
-  if (!module_invoke('ctools', 'api_version', OG_REQUIRED_CTOOLS_API)) {
+  if (!module_invoke('ctools', 'api_version', OG_SUBGROUP_REQUIRED_CTOOLS_API)) {
     return;
   }
 
